log4jSyslogWriter64k/src/main/java/com/github/loggly/log4j/helpers/SyslogTcpWriter64k.java

```diff
@@ -45,8 +47,12 @@ public void write(final char[] buf, final int off, final int len) throws IOExcep
 	public void write(final String string) throws IOException {
 		// compute syslog frame according to: https://tools.ietf.org/html/rfc6587
 		final String syslogFrame = String.format("%s %s", string.length(), string);
-
-		getWriter().append(syslogFrame);
+		try {
+			getWriter().append(syslogFrame);
+		} catch (final IOException e) {
+			close();
```

Comment:
This should probably be wrapped in a try-catch, ignoring IOExceptions coming from the close method to make sure, exceptions inside the close method do not hide the original IOException from here.